src/cpp/participant/implementations/auxiliar/DummyParticipant.cpp

```diff
@@ -24,14 +24,26 @@
 namespace eprosima {
 namespace ddsrouter {
 
-std::map<ParticipantId, std::shared_ptr<DummyParticipant>> DummyParticipant::participants_;
+std::mutex DummyParticipant::static_mutex_;
+std::map<ParticipantId, DummyParticipant*> DummyParticipant::participants_;
 
 DummyParticipant::DummyParticipant(
         ParticipantConfiguration participant_configuration,
         std::shared_ptr<DiscoveryDatabase> discovery_database)
     : EchoParticipant(participant_configuration, discovery_database)
 {
-    participants_[id()] = std::shared_ptr<DummyParticipant>(this);
+    participants_[id()] = this;
+}
+
+DummyParticipant::~DummyParticipant()
+{
+    std::unique_lock<std::mutex> lock(static_mutex_);
+
+    writers_.clear();
+    readers_.clear();
+
+    // participants_[id()] = nullptr;
```

Comment:
```suggestion
    writers_.clear();
    readers_.clear();

    std::unique_lock<std::mutex> lock(static_mutex_);
```

Reply:
Why? If the mutex is not taken, a new writer could be created before the destruction has finished.